crypto/engine/eng_cryptodev.c

```diff
@@ -1228,30 +1228,24 @@ static int bn2crparam(const BIGNUM *a, struct crparam *crp)
     crp->crp_p = (caddr_t) b;
     crp->crp_nbits = bits;
 
-    BN_bn2bin(a, b);
+    BN_bn2lebinpad(a, b, bytes);
```

Comment:
Presumably it doesn't make any difference if you use `BN_bn2lebin` or `BN_bn2lebinpad` here...or am I missing some subtlety? I also wonder about the use of "zalloc" above. Wouldn't "malloc" do?

Reply:
> Presumably it doesn't make any difference if you use `BN_bn2lebin` or `BN_bn2lebinpad` here...

I suspect the difference will be huge.  `BN_bn2lebinpad` exists...

> I also wonder about the use of "zalloc" above. Wouldn't "malloc" do?

Way back, it was a malloc followed by a zeroing (zeroizing?) memset.  That got replaced with a calloc, and subsequently a OPENSSL_zalloc.  Judging from b8555601922bce57077a9691cb3bcc31735ee300, it may have to do with some warning about uninitialized memory.

Reply:
> I suspect the difference will be huge. `BN_bn2lebinpad` exists...

Haha! That's a good enough reason! So we have `BN_bn2bin` and `BN_bn2binpad` but we only have `BN_bn2lebinpad`. Curious.

> Judging from [b855560](https://github.com/openssl/openssl/commit/b8555601922bce57077a9691cb3bcc31735ee300), it may have to do with some warning about uninitialized memory.

Hmmm. Not sure that still applies...but no big deal I guess.

Reply:
> So we have `BN_bn2bin` and `BN_bn2binpad` but we only have `BN_bn2lebinpad`. Curious.

The lebin functions are a late addition, and I assume that they were created on a use case basis rather than some idea of completeness.  Also, I do believe this is related to constant time (I haven't looked), so it's possible one might *not* want `BN_bn2lebin`...